src/appshell/view/dockwindow/internal/dockframemodel.cpp

```diff
@@ -95,7 +95,7 @@ void DockFrameModel::listenChangesInFrame()
         emit currentDockUniqueNameChanged();
     });
 
-    connect(qApp, &QApplication::aboutToQuit, [numDocksChangedCon, currentDockWidgetChangedCon]() {
+    connect(qApp, &QApplication::aboutToQuit, [numDocksChangedCon, currentDockWidgetChangedCon, this]() {
```

Comment:
I got my Windows laptop working again and tried myself; doesn't work. So either MSVC or Clang is wrong here, and in my opinion it's MSVC. 

Of course there is this terrible solution:
```cpp
    connect(qApp, &QApplication::aboutToQuit, [numDocksChangedCon, currentDockWidgetChangedCon
#if (defined (_MSCVER) || defined (_MSC_VER))
                                               , this
#endif
            ]() {
        disconnect(numDocksChangedCon);
        disconnect(currentDockWidgetChangedCon);
    });
```
Should we do this?

Reply:
Not sure. Doing this just to prevent a probably bogus compiler warning it not worth it IMHO
Maybe reverting my change and adding a comment in the code `// causes bogus compiler warning in MSVC` is the way to go?

Reply:
It using your solution, it should go along with a similar comment, `// prevent bogus compiler warning in MSVC`

Reply:
Actually go for that, including the comment. It is ugly, but could be even uglier...

Reply:
See #8798